app/src/main/java/com/realityexpander/tasky/agenda_feature/domain/AgendaItem.kt

```diff
@@ -40,11 +42,13 @@ abstract class AgendaItem {
         override val id: UuidStr,
         override val title: String,
         override val description: String,
-        val remindAt: ZonedDateTime,
         val isSynced: Boolean = false,
 
         val time: ZonedDateTime,
         override var startTime: ZonedDateTime = time, // for sorting in Agenda
+        val remindAt: ZonedDateTime,
+        override var remindAtTime: ZonedDateTime = remindAt,
```

Comment:
Whats the different between remindAt and remindAtTime?

Reply:
remindAt is represented as a `Long` in the DTO and Entity classes. So to make it clearer and easier to work with, i added a second property to leave the more human-readable form of the ZonedDateTime (for debugging)

## FOLLOW UP QUESTION 2

Let me know if there is a better approach.

Reply:
Yeah I agree that having 2 so similar properties with the same type is quite confusing 🤔 
I think one property is fully enough here and if you see `remindAt` is of type `ZonedDateTime` this makes it fully clear to me

Reply:
I guess Im not understanding what to do here.

I have a `remindAt` which is a `ZonedDateTime` in the domain layer, but its a `Long` represending `EpochSeconds` in the data layer. Also, when debugging the data layer, the `ZonedDateTime` are easier to work with (shows up as a human readable in the debugger) and are marked `@Transient`, so the data layer never stores it or sends it.

I hope this makes more sense why I am using two variables here.

Reply:
But in this case, both are of type ZonedDateTime

And it says: remindAtTime = remindAt

Thats what im not fully understanding

Reply:
Yes that's the same I don't understand. Nothing wrong with storing it as ZonedDateTime, but why have 2 same properties?

Reply:
OK, i could store them as ZonedDateTime strings.

The payload coming from the API is in `Epoch` `Long`

I want to be able to check if the date is correct and human readable, which is why I copy the `remindAt` time to a ZonedDateTime and keep it for debugging purposes. I was finding myself constantly using the web sites to convert `EpochMillis` to `ZonedDateTime` format, and this was an easy way to reduce that.

I hope this makes sense.

Reply:
It would be new to me that it could do that without type converters, but if you say it can compare the other way around why would it not be able to do it 🤔 
Either way, regarding the issue you usually have an entities model in data which contains the fields stored in your DB and you map that to your domain model when reading it. Why does the domain model then need 2 date time fields?

Reply:
I need two because one of the fields (the `EpochSeconds Long` and `EpochMillis Long`) are not human readable, and I needed to use a conversion app to see what the dates and times actually are. So instead of having to use the conversion website to see what the dates are for these `Longs`, I just used another var of type `ZoneDateTime` so its very easy to see what the date is without resorting to using an app.

How do you convert `long` `EpochMillis` and `EpochSeconds` to human readable when you are debugging?

Reply:
If I just want to check which date is arriving from the API, I inspect the dateTime value of the domain model to see that.  I think what would make more sense in this case is to give `AgendaItem` a constructor and put the abstract values in there (without abstract). That way, the child classes only have one field per type.
It's quite confusing if you see `remindAt` and `remindAtTime` as properties for an event, since they're effectively the same.

Reply:
How do you overload the data stored in the Room Database? AFAIK you can only store a single type in the Entity. Pretty confused about what is being suggested here.

For dates and times, The domain layer uses `ZonedDateTime`. The data layer (room) MUST use `long` to represent the date/time. This is what is stored in the database and what Room uses for comparisons using SQL queries.

Please clarify about what is to be stored where and what types:

ie: data layer -> `long` representing Epoch Seconds for database, `long` representing Epoch Millis for the API
     domain layer -> `ZonedDateTime`
     presentation layer -> `ZonedDateTime`

Im missing something important here, and cant figure out what it is yet.

Reply:
"Using the above setup, it's not possible to visually just look at the data and know the date using the Database inspector by looking at the raw data at the column. Same for the API calls looking at the raw data in and out. You must rely on the domain layer to see the data is correct. If this is the way it works, then I will leave it be."

Since the API and DB use epoch seconds (Long), then no, a human won't fully understand the date just by looking at it, and you'd need to use a site like https://www.epochconverter.com/ to understand it. That's why as soon as you receive it from either DB or API, you convert it to a ZonedDateTime on your Domain model. So it's easy to read and understand

You should only use Long (seconds) when storing to API and DB, and the reason why is because if you store as Date, each person understands date a different way. That's why there are many ways to say a specific date (month/day/year, day/month/year, day-month-year, etc). But if you just store it as seconds, then when you parse it to ZonedDateTime, you know the date is 100% correct

Reply:
OK, this is good confirmation. Thank you!